### ISSUE TITLE
Auto-complete keeps inserting a 'd' when trying to change rate duration

### ISSUE THREAD

**Issue by modelbitjason** [prometheus/prometheus#15452]
### What did you do?

1. Graph a rate `rate(foo_count[5m])`
2. Attempt to graph a different duration
3. 'd' is always inserted by autocomplete

### What did you expect to see?

A rate with say 1 or 10 minute buckets not an error.

Ideally the autocomplete would see that my expression would be more valid if it did nothing rather than creating a broken duration.

### What did you see instead? Under which circumstances?

An error message unless I also press escape first, then enter.

[prom_bug.webm](https://github.com/user-attachments/assets/8bc4dc1d-c1f0-409e-96cd-65638aa72d67)

### System information

Chrome on linux 129.0.6668.70 (Official Build) (64-bit) 

### Prometheus version

```text
{"version":"3.0.0","revision":"c5d009d57fcccb7247e1191a0b10d74b06295388","branch":"HEAD","buildUser":"root@aa286d0eb00a","buildDate":"20241114-16:40:43","goVersion":"go1.23.3"}
```

**Comment by juliusv** [comment]
cc @Nexucis

**Comment by anshgoyalevil** [comment]
I see @Nexucis has been assigned to this issue. I am open to work upon it if assigned person hasn't started the work

**Comment by bboreham** [comment]
Hello from the bug-scrub!  On https://prometheus.demo.prometheus.io/query we can see the symptom, so long as we hit Enter after autocomplete comes up.  @modelbitjason can you confirm that's what you did?

@Nexucis do you have any thoughts?

**Comment by adity-a34** [comment]
Hi @juliusv @Nexucis @bboreham 
I have encountered the same behavior as @bboreham described. i.e. autocomplete does not append `d` automatically as long as `Enter` key is not hit.
Given the issue is open @juliusv What fix do you expect?
I guess the suggestion of additional `d/m` is not to be given as long as either `d` or `m` is present next to a digit/integer.

**Comment by juliusv** [comment]
> Given the issue is open [@juliusv](https://github.com/juliusv) What fix do you expect? I guess the suggestion of additional `d/m` is not to be given as long as either `d` or `m` is present next to a digit/integer.

Yeah, I would expect us to notice in this completion context that the current duration node we're in already has a unit suffix, in which case we shouldn't try to complete another one.

Currently of course you can press Esc (to close the completion dropdown) and then hit Enter to get the desired result, but that's not great.
